packages/job-worker/src/blueprints/context/adlibActions.ts

```diff
@@ -102,6 +103,10 @@ export class ActionExecutionContext extends ShowStyleUserContext implements IAct
 		super(contextInfo, _context, showStyle, watchedPackages)
 	}
 
+	async getUpcomingParts(limit: number = 5): Promise<ReadonlyDeep<IBlueprintPart[]>> {
+		return getOrderedPartsAfterPlayhead(this._context, this._playoutModel, limit)
```

Comment:
Both of these should be running the parts through `convertPartToBlueprints`, to sanitise/clone them and make them safe for blueprints.

I'm a  little surprised that typescript didn't flag this, I thought that some of the properties were incompatible.. weird